### PR TITLE
fix: refactor json compact error log

### PR DESCRIPTION
**Pull request by Allen8975701**

### CODE REVIEW

**Comment by Allen8975701** [comment]
currently, we can only get log like this "Fail to compact json: invalid character 'u' after top-level value"
this does not make sense and can't help to improve API

**Comment by Allen8975701** [comment]
@marselab @wardab0104 @angelo-chan

**Review by marselsampe — APPROVED** [review]

File: pkg/logger/log/accesslog.go [hunk 1]
```diff
@@ -265,7 +265,7 @@ func getResponseBody(respWriter *ResponseWriterInterceptor, contentType string)
 	}
 
 	if strings.Contains(contentType, "application/json") {
-		return util.MinifyJSON(respWriter.data)
+		return util.MinifyJSON(respWriter.data, "")
```
Comment (marselsampe) [inline pkg/logger/log/accesslog.go:268]: We could get the API URL from `requestUri` in parent method

Screenshot from 2024-06-13 11-27-39

File: pkg/util/jsonhelper.go [hunk 2]
```diff
@@ -17,22 +17,23 @@ package util
...
+func MinifyJSON(jsonBytes []byte, details string) string {
 	buffer := new(bytes.Buffer)
 	if err := json.Compact(buffer, jsonBytes); err != nil {
-		logrus.Warnf("Fail to compact json: %v", err)
-
 		// the uncompleted json will throw an error, i.e. {"foo":"bar"
 		// so instead, switch with the simple parse (e.g. removing new lines)
 		jsonString := string(jsonBytes)
+		// add this log to check which request/response body is not with json format
+		logrus.Warnf("Fail to compact json: %v, source : %s, details: %s", err, jsonString, details)
```
Comment (marselsampe) [inline pkg/util/jsonhelper.go:32]: Better to indicate if it's from request or response,
e.g. `Fail to compact request json` or `Fail to compact response json`

Comment (marselsampe) [inline pkg/util/jsonhelper.go:32]: Should we print it after removing the new line character from `jsonString`?

File: pkg/logger/log/accesslog.go (hunk 1 above)
Comment (Allen8975701) [reply pkg/logger/log/accesslog.go]: > We could get the API URL from `requestUri` in parent method
> 
> Screenshot from 2024-06-13 11-27-39
updated

File: pkg/util/jsonhelper.go (hunk 2 above)
Comment (Allen8975701) [reply pkg/util/jsonhelper.go]: updated @marselsampe

Comment (Allen8975701) [reply pkg/util/jsonhelper.go]: updated. previously I thought  we need print the original input value & format. @marselsampe

**Review by marselsampe — APPROVED** [review]

File: pkg/util/jsonhelper.go [hunk 3]
```diff
@@ -17,22 +17,24 @@ package util
...
 	"encoding/json"
-	"strings"
-
 	"github.com/sirupsen/logrus"
+	"strings"
 )
 
 // MinifyJSON is used to compact the json bytes (e.g. removing whitespaces and new lines character)
-func MinifyJSON(jsonBytes []byte) string {
+func MinifyJSON(jsonBytes []byte, from, details string) string {
```
Comment (angelo-chan) [inline pkg/util/jsonhelper.go:25]: Actually, I don't like the method abstract here:
1. minifyJSON is a purely least unit, it shouldn't involve any business logic, don't add extra params like details, don't log in the method, instead throw the error directly
2. It's the consumer's responsibility to log and track the detail

File: pkg/logger/log/accesslog.go [hunk 4]
```diff
@@ -225,7 +225,7 @@ func AccessLog(req *restful.Request, resp *restful.Response, chain *restful.Filt
 }
 
 // getRequestBody will get the request body from Request object
-func getRequestBody(req *restful.Request, contentType string) string {
+func getRequestBody(req *restful.Request, contentType, requestURL string) string {
```
Comment (angelo-chan) [inline pkg/logger/log/accesslog.go:228]: Why must we consider whether it's a request or a response? Shouldn't we focus on the bytes directly?

Comment (Allen8975701) [reply pkg/logger/log/accesslog.go:228]: updated

**Review by angelo-chan — APPROVED** [review]

**Pull request merged**